### PR TITLE
fr.janssoone.pkgtransporter

### PR DESCRIPTION
**Pull request by knacky34**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - I'm requiring access to flatpak directories to be able to list applications then backup them
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: fr.janssoone.pkgtransporter.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+    "app-id" : "fr.janssoone.pkgtransporter",
```
Comment (hfiguiere) [inline fr.janssoone.pkgtransporter.json:2]: is `janssoone.fr` really controlled by the upstream developer ?

File: fr.janssoone.pkgtransporter.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+	"--filesystem=/var/lib/flatpak/app:ro",
+        "--filesystem=xdg-data:ro",
+        "--filesystem=xdg-config:ro",
+        "--filesystem=~/.var/app/:ro"
```
Comment (hfiguiere) [inline fr.janssoone.pkgtransporter.json:16]: some of these filesystem permission are not allowed by default.

File: fr.janssoone.pkgtransporter.json (hunk 1 above)
Comment (knacky34) [reply fr.janssoone.pkgtransporter.json:2]: Non this romain don't exist, I've misunderstood.
Could replace it with com.gitlab ?

File: fr.janssoone.pkgtransporter.json (hunk 2 above)
Comment (knacky34) [reply fr.janssoone.pkgtransporter.json:16]: The app need access to ~/.var and /var, at least ~/.var
How can I allow theses paths ?

File: fr.janssoone.pkgtransporter.json (hunk 1 above)
Comment (hfiguiere) [reply fr.janssoone.pkgtransporter.json:2]: see
https://docs.flatpak.org/en/latest/conventions.html#application-ids

linked from
https://docs.flathub.org/docs/for-app-authors/requirements#application-id

File: fr.janssoone.pkgtransporter.json (hunk 2 above)
Comment (hfiguiere) [reply fr.janssoone.pkgtransporter.json:16]: lines 13-16.

xdg-data and xdg-config are not allowed by default either.

https://docs.flathub.org/docs/for-app-authors/linter

File: fr.janssoone.pkgtransporter.json (hunk 1 above)
Comment (knacky34) [reply fr.janssoone.pkgtransporter.json:2]: Thanks,
I've changed the appId to `io.gitlab.gwendalj.package-transporter`.

File: fr.janssoone.pkgtransporter.json (hunk 2 above)
Comment (knacky34) [reply fr.janssoone.pkgtransporter.json:16]: I understand how to add exceptions for `xdg-*` directories but I can't find the linter error for custom paths.

Comment (hfiguiere) [reply fr.janssoone.pkgtransporter.json:16]: they are not blocked by the linter.

File: fr.janssoone.pkgtransporter.json (hunk 1 above)
Comment (knacky34) [reply fr.janssoone.pkgtransporter.json:2]: Not yet, I was waiting the answer for the other comment before pushing this change

File: fr.janssoone.pkgtransporter.json (hunk 2 above)
Comment (knacky34) [reply fr.janssoone.pkgtransporter.json:16]: Okay so I don't need exceptions for this two paths ?

File: fr.janssoone.pkgtransporter.json
[the remark this answers is not in the document]
Comment (knacky34) [reply fr.janssoone.pkgtransporter.json:1]: The manifest already have this name.
[see project repo](https://gitlab.com/GwendalJ/package-transporter/-/blob/master/io.gitlab.gwendalj.package-transporter.json?ref_type=heads)

[the remark this answers is not in the document]
Comment (knacky34) [reply fr.janssoone.pkgtransporter.json:1]: It's fixed

**Comment by hfiguiere** [comment]
bot, build io.gitlab.gwendalj.package-transporter

**Comment by knacky34** [comment]
It should be fixed now.

**Comment by knacky34** [comment]
I'm completing the appdata.xml

**Comment by knacky34** [comment]
bot, build io.gitlab.gwendalj.package-transporter

**Comment by knacky34** [comment]
bot, build io.gitlab.gwendalj.package-transporter

File: io.gitlab.gwendalj.package-transporter.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        "/share/doc",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+        {
+          "name": "blueprint-compiler",
```
Comment (hfiguiere) [inline io.gitlab.gwendalj.package-transporter.json:29]: there should be a cleanup rule to remove it all since it's only a build time dependency

Comment (hfiguiere) [reply io.gitlab.gwendalj.package-transporter.json:29]: https://github.com/flathub/io.posidon.Paper/blob/79c31579c9e0ae987357410864aa6ed6372d9bad/io.posidon.Paper.json#L42

**Pull request closed without merging**